test/Prototypes/Unicode.swift

```diff
@@ -6,6 +6,42 @@ import Swift
 import SwiftShims
 import StdlibUnittest
 
+protocol Unicode {
+  associatedtype Encoding: UnicodeEncoding
+  associatedtype CodeUnits: RandomAccessCollection
+  /* where CodeUnits.Iterator.Element == Encoding.CodeUnit */
+  var codeUnits: CodeUnits {get}
+
+  associatedtype ValidUTF8View : BidirectionalCollection
+  // where ValidUTF8View.Iterator.Element == UTF8.CodeUnit */
+  // = TranscodedView<CodeUnits, Encoding, UTF8>
+  var utf8: ValidUTF8View {get}
+
+  associatedtype ValidUTF16View : BidirectionalCollection
+  // where ValidUTF16View.Iterator.Element == UTF16.CodeUnit
```

Comment:
@milseman I originally wanted to comment that it's in the tests, but then thought that eventually  this code has a non-0 chance of being copied in the final implementation, so it would be nice to not think about including the `FIXME(ABI)` at that point and risk forgetting it...